### DISCUSSION TITLE
Tailwind JIT + Gulp possible?

### DISCUSSION THREAD

**Discussion by jenstornell** [tailwindlabs/tailwindcss#4182] [Help] [answered]
I've not tried out JIT yet but watched Youtube videos about it. It seems like it runs like a watcher, waiting for a file to be saved.

I use Gulp to compress and combine javascript files. It is also a watcher that is waiting for files to be saved, in the same CLI.

**Is it possible to run Tailwind JIT and Gulp at the same time?**

**Comment by adamwathan** [comment]
Should be possible, what have you tried and what issues are you seeing? If you can create a repository with a basic setup in place that isn't working I can look at it.

**Reply by jenstornell** [reply]
## Repo

Alright! I've created a repo. I can run JIT or Gulp without any problems, but not at the same time. I use NPM.

https://github.com/jenstornell/tailwind-jit-gulp

## JIT

I'm not an NPM expert so in order to understand how to install JIT, I used this Youtube tutorial: https://youtu.be/lYWQdxHGnkc which worked fine.

**To install JIT dependencies:**

```text
npm install -D tailwindcss autoprefixer postcss-cli @tailwindcss/jit
```

**To run JIT:**

```text
npm run build
```

When watcher has started change and save something in `layouts/index.html`. Then it updates `assets/css/dist/tailwind.purged.css`.

## Gulp

**To install the gulp dependencies:**

```text
npm install --save-dev gulp gulp-concat gulp-uglify-es gulp-rename
```

**To run Gulp:**

```text
gulp
```

When watcher has started change and save something in `assets/js/src/script.js`. Then it updates `assets/js/dist/script.min.js`.

## The problem

When I start JIT with `npm run build` I see a watcher in the terminal which prints out:

```text
Processing assets\css\src\tailwind.css...
Finished assets\css\src\tailwind.css in 7.28 s

Waiting for file changes...
```

It works fine but to run Gulp, I need to `CTRL + C` (Windows) to break this watcher.

```text
Terminate batch job (Y/N)?
```

Then I can run `gulp`:

```text
[11:13:04] Using gulpfile D:\projects\tools\tailwind-jit-gulp\gulpfile.js
[11:13:04] Starting 'default'...
```

To break a watcher to start another is not a very good workflow in my opinion.

## Possible solutions?

- Somehow bundle both watcher into one multiwatcher
- Use two terminals and run them simultaneously
- Switch between them all the time (not optimal workflow)

**Comment by jenstornell** [comment, marked as answer]
I've found a solution. In VSCode it's possible to use two terminals at the same time. I'm not sure if it's the best one but I'll use it for now.

![split](https://user-images.githubusercontent.com/927011/116355162-bf1d5d00-a7f9-11eb-933e-e3314f7ad86c.png)
